### PR TITLE
Add detailed installation and configuration guide

### PR DESCRIPTION
**Pull request by ianopolous**

### CODE REVIEW

**Comment by ianopolous** [comment]
FYI: Currently the go version used to build go-ipfs is not published with releases. So people currently don't have a way to build this without inside knowledge.

**Comment by Stebalien** [comment]
> FYI: Currently the go version used to build go-ipfs is not published with releases. So people currently don't have a way to build this without inside knowledge.

It's published in https://dist.ipfs.io/go-ipfs/v0.4.22/build-info but not in a machine readable format. You can also get it by running `ipfs version --all`.

Really, we _could_ write a script to build a plugin against the installed IPFS version:

1. Run `ipfs version --all --enc=json` to get the architecture information and go version.
2. Download go if necessary.
3. Use `ipfs version deps` to get a list of go-ipfs deps and _their_ versions. We can then feed that into go mod (like how the current `set-target.sh` script works).

But that's a lot of black magic.

File: README.md [hunk 1]
```diff
@@ -21,6 +22,63 @@ To update the go-ipfs, run:
 ```bash
 > make go.mod IPFS_VERSION=version
 ```
+## Detailed Installation
+For an brand new ipfs instance (no data stored yet)
```
Comment (Stebalien) [inline README.md]:
```suggestion
For a brand new ipfs instance (no data stored yet)
```

File: README.md [hunk 2]
```diff
@@ -21,6 +22,63 @@ To update the go-ipfs, run:
...
+            "secretKey": ""
+          },
+          "mountpoint": "/blocks",
+          "prefix": "s3.datastore",
+          "type": "measure"
+        },
+```
+
+If you are configuring a brand new ipfs instance without any data, you can overwrite the datastore_spec file with:
+> {"mounts":[{"bucket":"peergos-test","mountpoint":"/blocks","region":"us-east-1","rootDirectory":""},{"mountpoint":"/","path":"datastore","type":"levelds"}],"type":"mount"}
```
Comment (Stebalien) [inline README.md]: nit: mentions peergos-test

Comment (ianopolous) [reply README.md]: changed to $bucketname

File: README.md [hunk 3]
```diff
@@ -21,6 +22,63 @@ To update the go-ipfs, run:
 ```bash
 > make go.mod IPFS_VERSION=version
 ```
+## Detailed Installation
+For a brand new ipfs instance (no data stored yet)
+1. copy s3plugin.so $IPFS_DIR/plugins/go-ds-s3.so
```
Comment (Stebalien) [inline README.md]: One should be able to run:

1. `ipfs init`.
2. `make install IPFS_VERSION=$VERSION`.

That will install the plugin into the IPFS repo (and fix permissions, etc.).

**Comment by Stebalien** [comment]
Thanks!

File: README.md (hunk 3 above)
Comment (ianopolous) [reply README.md]: Yep, but I'm not installing locally and thought this would be useful to know regardless. I can make the local option clearer

Comment (ianopolous) [reply README.md]: the next thing I'm doing is to automate this whole process

Comment (Stebalien) [reply README.md]: Awesome! Then look at https://github.com/ipfs/go-ds-s3/pull/25#issuecomment-521428605. We should be able to take an ipfs binary and build a matching plugin.

Comment (ianopolous) [reply README.md]: Sorry I meant the install process, not the building. (to go with our current IPFS installer and runtime manager) - 
https://github.com/Peergos/Peergos/blob/master/src/peergos/server/storage/IpfsInstaller.java
https://github.com/Peergos/Peergos/blob/master/src/peergos/server/storage/IpfsWrapper.java

**Pull request merged**